src/main.ts

```diff
@@ -563,6 +574,15 @@ async function getHashForBlock(
 
 		return await api.rpc.chain.getBlockHash(blockNumber);
 	} catch (err) {
+		// Check if the block number is too high
+		const { number } = await api.rpc.chain.getHeader();
+		if (blockNumber && number.toNumber() < blockNumber) {
```

Comment:
Since you put `let blockNumber` before the `try`, will it always consider this truthy?

Reply:
On line 549 `blockNumber` is equal to `undefined` - so it is a falsey value